Library/Homebrew/language/python.rb

```diff
@@ -285,10 +285,44 @@ def pip_install_and_link(targets)
 
         def do_install(targets)
           targets = Array(targets)
+          targets.map! { |target| substitute_poetry_wheel_if_needed(target) }
+
           @formula.system @venv_root/"bin/pip", "install",
                           "-v", "--no-deps", "--no-binary", ":all:",
                           "--ignore-installed", *targets
         end
+
+        # Generates a wheel from `poetry build` for Poetry-based targets, if
+        # needed, which can then be consumed by `pip install`.
+        #
+        # @param target [String, Pathname] the path to the target
+        # @return target [Pathname] the path to a replacement wheel target,
+        #   or the original input target if wheel generation is not necessary
+        def substitute_poetry_wheel_if_needed(target)
+          target_as_path = Pathname.new(target)
+          return target unless target_needs_poetry?(target_as_path)
+
+          old_files = Dir[target_as_path/"dist/*.whl"].to_set
+          @formula.system Formula["poetry"].opt_bin/"poetry", "build", "-vvv", "--format", "wheel"
+          new_files = Dir[target_as_path/"dist/*.whl"].to_set
+
+          wheels = new_files - old_files
+          wheels.to_a.first
```

Comment:
Why do we only care about a single result here?

Reply:
I would expect there to only ever be one new file in that difference of sets (the Python ecosystem is pretty consistent about one wheel/file generated per package - if its not an actual PEP somewhere, then it's certainly at least a de-facto standard). I don't know if there's a more elegant/idiomatic way to convey that assumption here - would a comment help?

Reply:
> I would expect there to only ever be one new file in that difference of sets (the Python ecosystem is pretty consistent about one wheel/file generated per package - if its not an actual PEP somewhere, then it's certainly at least a de-facto standard). I don't know if there's a more elegant/idiomatic way to convey that assumption here - would a comment help?

I think it would be good to `raise` if there's a state here that's unexpected e.g. zero or multiple files?